engine/runtime-parser/src/test/java/org/enso/compiler/core/EnsoParserTest.java

```diff
@@ -1246,6 +1246,40 @@ public void testSkip() throws Exception {
     equivalenceTest("a = x", "a = SKIP FREEZE x.f y");
   }
 
+  @Test
+  public void testBlockSyntax() throws Exception {
+    equivalenceTest("""
+    nums v fm ff n = v . map fm . filter ff . take n
+    """, """
+    nums v fm ff n = v
+        . map fm
+        . filter ff
+        . take n
+    """);
+  }
+
+  @Test
+  public void testBlockSyntaxOperators() throws Exception {
+    equivalenceTest("""
+    value = nums * each random + constant
+    """, """
+    value = nums
+        * each random
+        + constant
+    """);
+  }
+
+  @Test
+  public void testBlockSyntaxOperators2() throws Exception {
+    equivalenceTest("""
+    value = (nums + each random) * constant
+    """, """
+    value = nums
+        + each random
+        * constant
+    """);
```

Comment:
According to @somebody1234's [comment](https://github.com/enso-org/enso/pull/8415/files#r1408921830) this is the desired behavior.